### PR TITLE
🐛 svg styles override unintentionally

### PR DESCRIPTION
**Pull request by arunachalam-monk**
## Overview
Jira Ticket Reference : [MN-554](https://acvauctions.atlassian.net/browse/MN-554)
seems `getAttributes` is used to add styles it unintentionally overrides the style.

[MN-554]: https://acvauctions.atlassian.net/browse/MN-554?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: documentation/src/components/Sights/SightCard/SightCard.tsx [hunk 1]
```diff
@@ -53,11 +54,13 @@ export function SightCard({ item }: SightCardProps) {
     copyPopupRef.current?.open();
   };
 
-  const getOverlayAttributes = () => ({
-    style: {
-      stroke: isDarkTheme ? '#ffffff' : '#000000',
-    },
-  });
+  const getOverlayAttributes: DynamicSVGCustomizationFunctions['getAttributes'] = () => {
+    return {
```
Comment (souyahia-monk) [inline documentation/src/components/Sights/SightCard/SightCard.tsx:58]: replace `return` statement with direct arrow function return

File: documentation/src/components/Sights/TopBar/TopBar.tsx [hunk 2]
```diff
@@ -2,13 +2,13 @@ import { labels, sights, vehicles } from '@monkvision/sights';
...
 import clsx from 'clsx';
-import React, { ReactElement, useRef } from 'react';
+import React, { FunctionComponentElement, useRef } from 'react';
 import styles from './TopBar.module.css';
 
 export interface ListItem {
   id: string;
   lookups: string[];
-  component: ReactElement;
+  component: FunctionComponentElement<typeof SightCard>;
```
Comment (souyahia-monk) [inline documentation/src/components/Sights/TopBar/TopBar.tsx:11]: Why did you change this exactly? I think `component` could be any type of displayable element. Not ONLY a `SightCard` component

**Comment by souyahia-monk** [comment]
Also : you forgot to add a test case for this PR

File: documentation/src/components/Sights/TopBar/TopBar.tsx (hunk 2 above)
Comment (arunachalam-monk) [reply documentation/src/components/Sights/TopBar/TopBar.tsx:11]: Technically it can be any component we wanted. but we are only using SightCard from `pages/Sights/index.tsx` its not clear. We need to go and look into other files to find what are the components we are using. 
Now we can see that in index.tsx file itself we can see the component using

Comment (souyahia-monk) [reply documentation/src/components/Sights/TopBar/TopBar.tsx:11]: I think we just need to think about the objective of the ListItem object. What is its purpose? It is just an item in the list that contains a `component` property that is any React element to be displayed on the screen that's it. So let's just make it like that. I see no advantage of strongly typing the `component` property

File: packages/common-ui-web/src/components/DynamicSVG/hooks/useJSXTransformAttributes.ts [hunk 3]
```diff
@@ -42,8 +42,8 @@ export function useJSXTransformAttributes(element: Element) {
         return {
           ...prev,
           [key]: value,
-        };
-      }, {}),
+        } as Partial<JSX.IntrinsicElements>;
+      }, {}) as JSX.IntrinsicElements,
```
Comment (souyahia-monk) [inline packages/common-ui-web/src/components/DynamicSVG/hooks/useJSXTransformAttributes.ts:46]: Remove this type cast and instead add a generic type to reduce :

```typescript
    return useMemo(
    () =>
      element.getAttributeNames().reduce<Partial<JSX.IntrinsicElements>>((prev, attr) => {
        const { key, value } = tranformJsxAttribute({
          key: attr,
          value: element.getAttribute(attr),
        });
        return {
          ...prev,
          [key]: value,
        };
      }, {}),
    [element],
  );
```

File: documentation/src/components/Sights/TopBar/TopBar.tsx [hunk 4]
```diff
@@ -21,8 +21,7 @@ export interface TopBarProps {
   onSelectTab: (tab: Tab) => void;
   onLookupInput: (value: string) => void;
 }
-
-export const tabs = {
+export const tabs: Record<'sights' | 'vehicles' | 'labels', Tab> = {
```
Comment (souyahia-monk) [inline documentation/src/components/Sights/TopBar/TopBar.tsx:24]: Remove this type declaration : let TypeScript infer the type

File: packages/common-ui-web/test/components/DynamicSVG/hooks/useCustomAttributes.test.ts [hunk 5]
```diff
@@ -7,50 +7,63 @@ describe('useCustomAttributes hook', () => {
...
     const { result, unmount } = renderHook(useCustomAttributes, {
-      initialProps: { element, groupIds, getAttributes },
+      initialProps: { element, groupIds, getAttributes, style: customAttr.style },
     });
 
     expect(getAttributes).toHaveBeenCalledWith(element, groupIds);
     expect(result.current).toEqual(customAttr);
     unmount();
   });
+  it('should apply style correctly', () => {
```
Comment (souyahia-monk) [inline packages/common-ui-web/test/components/DynamicSVG/hooks/useCustomAttributes.test.ts:56]: blank line above `it` statement

File: documentation/src/components/Sights/TopBar/TopBar.tsx (hunk 4 above)
Comment (arunachalam-monk) [reply documentation/src/components/Sights/TopBar/TopBar.tsx:24]: It's inferring the types only while we use the tabs value not putting any restrictions while writing the code.
for example.
```ts
const tabs = {/** the tab value is wrong */} // won't tell error on hear
const a: Tab = tabs.sight; // will tell error.
```

The main problem is if someone adds a new property for one of the tabs and uses it we won't be able to find it on code review. after 10 iterations on the tabs value the things will be out of track.

File: packages/common-ui-web/test/components/DynamicSVG/hooks/useCustomAttributes.test.ts (hunk 5 above)
Comment (souyahia-monk) [reply packages/common-ui-web/test/components/DynamicSVG/hooks/useCustomAttributes.test.ts:56]: @arunachalam-monk this is not fixed
